src/main/java/eu/europeana/metis/sandbox/domain/Record.java

```diff
@@ -4,29 +4,40 @@
 
 import eu.europeana.metis.sandbox.common.locale.Country;
 import eu.europeana.metis.sandbox.common.locale.Language;
+import java.nio.charset.Charset;
+import java.nio.charset.StandardCharsets;
+import java.util.Arrays;
 import java.util.Objects;
 import java.util.StringJoiner;
 
 /**
- * Immutable object that represents a record
+ * Object that represents a record.
+ * <br /><br />
+ * Stores the record content as a byte[], do not
+ * override the byte[] contents after object construction, we are not making a copy of it because it
+ * is expensive and Record object is expected to be use as a non mutable object.
  */
 public class Record {
 
+  private static final Charset DEFAULT_CHARSET = StandardCharsets.UTF_8;
+
   private final String recordId;
   private final String datasetId;
   private final String datasetName;
   private final Country country;
   private final Language language;
-  private final String content;
+  private final byte[] content;
```

Comment:
Now content is a `byte[]`. And while still preserve the String, it is created from the provided `byte[]`.
But now that this field is a `byte[]` it is no longer an immutable object, I thought of returning a copy of the `content` but then it will be expensive. So I decided to make a note on not modyfing the content byte array.
On the other hand it is now easier to pass the events through the message broker since the content is now a byte array, but we still need the String for some steps that take a String in the parameter

Reply:
Please let us know which steps require a string! Perhaps internally it is converted to a byte array again and it would be best to expose a method that accepts a byte array.

Reply:
I checked some steps and you are right, I saw libraries converting to a byte array again, but even in that case they do not return a byte array as a result (although you can probably get the bytes of the String they return). 
At first sight does not look trivial to do this, so I am taking your suggestion I will create a separate ticket for this.